robots/pkg/kubevirt/cmd/copy/jobs_test.go

```diff
@@ -151,87 +151,89 @@ func Test_copyPeriodicJobsForNewProvider(t *testing.T) {
 			want to report in any case, this empty config is preserved in the resulting modified version that has been
 			written to storage.
 		*/
-		{
-			name: "reporterconfig with empty job states slice is preserved even with no job state to report",
-			args: args{
-				jobConfig: &config.JobConfig{
-					Periodics: []config.Periodic{
-						{
-							JobBase: config.JobBase{
-								Labels: map[string]string{},
-								ReporterConfig: &v1.ReporterConfig{
-									Slack: &v1.SlackReporterConfig{
-										JobStatesToReport: []v1.ProwJobState{},
-									},
-								},
-								Name: prowjobconfigs.CreatePeriodicJobName(semver("1", "21", "0"), "sig-network"),
-								Spec: &corev1.PodSpec{
-									Containers: []corev1.Container{
-										{
-											Env: []corev1.EnvVar{},
-										},
-									},
-								},
-							},
-							Interval: "",
-							Cron:     "0 1,9,17 * * *",
-							Tags:     nil,
-						},
-					},
-				},
-				targetProviderReleaseSemver: semver("1", "22", "0"),
-				sourceProviderReleaseSemver: semver("1", "21", "0"),
-			},
-			wantUpdated: true,
-			wantJobConfig: &config.JobConfig{
-				Periodics: []config.Periodic{
-					{
-						JobBase: config.JobBase{
-							Labels: map[string]string{},
-							Name:   prowjobconfigs.CreatePeriodicJobName(semver("1", "21", "0"), "sig-network"),
-							ReporterConfig: &v1.ReporterConfig{
-								Slack: &v1.SlackReporterConfig{
-									JobStatesToReport: []v1.ProwJobState{},
-								},
-							},
-							Spec: &corev1.PodSpec{
-								Containers: []corev1.Container{
-									{
-										Env: []corev1.EnvVar{},
-									},
-								},
-							},
-						},
-						Interval: "",
-						Cron:     "0 1,9,17 * * *",
-						Tags:     nil,
-					},
-					{
-						JobBase: config.JobBase{
-							Annotations: map[string]string{},
-							Labels:      map[string]string{},
-							Name:        prowjobconfigs.CreatePeriodicJobName(semver("1", "22", "0"), "sig-network"),
-							ReporterConfig: &v1.ReporterConfig{
-								Slack: &v1.SlackReporterConfig{
-									JobStatesToReport: []v1.ProwJobState{},
-								},
-							},
-							Spec: &corev1.PodSpec{
-								Containers: []corev1.Container{
-									{
-										Env: []corev1.EnvVar{},
-									},
-								},
-							},
-						},
-						Interval: "",
-						Cron:     "10 2,10,18 * * *",
-						Tags:     nil,
-					},
-				},
-			},
-			wantJobStatesToReportInSerialization: true,
-		},
+		// TODO it seems that job_states_to_report has gotten an omitempty which effectively removes it when serializing
```

Comment:
Can we open an issue for this?

Reply:
Yes: https://github.com/kubevirt/project-infra/issues/3901